### PR TITLE
Validate DOB is not in the future; only send full SSN to MCI

### PR DESCRIPTION
**Pull request by gigxz**
Some fixes from more thorough MCI clearance QA.
* MCI api rejects incomplete SSNs, so don't send them
* MCI also rejects future DOBs, so dont send that, and dont allow it as input
* Reload record before returning from SubmitForm so that newly created MCI ID is immediately available

### CODE REVIEW

**Review by eanders — APPROVED** [review]
Looks good!

File: drivers/hmis/app/models/hmis/hud/validators/client_validator.rb [hunk 1]
```diff
@@ -17,6 +17,12 @@ def configuration
     Hmis::Hud::Client.hmis_configuration(version: '2022').except(*IGNORED)
   end
 
+  def self.hmis_validate(record, options: {}, **_)
+    errors = HmisErrors::Errors.new
+    errors.add :dob, :out_of_range, severity: :error, message: future_message, **options if record.dob&.future?
```
Comment (ttoomey) [inline drivers/hmis/app/models/hmis/hud/validators/client_validator.rb:22]: I wonder if we should also check for records in the distant past (a DOB before 1900?)

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/mci_payload.rb [hunk 2]
```diff
@@ -18,8 +18,8 @@ def self.from_client(client, mci_id: nil)
         'middleName' => client.middle_name,
         'lastName' => client.last_name,
         'suffix' => client.name_suffix,
-        'ssn' => client.ssn,
-        # 'ssnAlias' => client.ssn,
+        # API only accepts full SSNs
+        'ssn' => client.ssn&.match?(/^\d{9}$/) ? client.ssn : nil,
```
Comment (ttoomey) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/mci_payload.rb:22]: If the ssn is potentially dirty, perhaps add a method on client to strip out hyphens, spaces, etc.

```
def normalized_ssn
  result = ssn&.gsub(/[^0-9]/, '')
  result && result.size == 9 ? result : nil
end
```

**Review by ttoomey — APPROVED** [review]
looks good!

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/mci_payload.rb (hunk 2 above)
Comment (gigxz) [reply drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/mci_payload.rb:22]: It should already be clean at this point, HMIS specs allows partial ssns though by replacing missing numbers with "x" or "X"

File: drivers/hmis/app/models/hmis/hud/validators/client_validator.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/app/models/hmis/hud/validators/client_validator.rb:22]: Yeah good point, I added that

**Pull request merged**